### PR TITLE
Feat/ytsubcountmodule

### PR DESCRIPTION
**Pull request by SIMULATAN**

### CODE REVIEW

**Comment by SIMULATAN** [comment]
@JKamsker please fix the conflict, I just need to add the "YouTube" property in the appsettings.json. Thanks!

File: DiscordBot.Modules/Services/YoutubeService.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                return;
+            }
+            if (_channel2 == null)
+            {
+                return;
+            }
+
+            if (!(_guild.GetChannel(youtubeSettings.ChannelID2) is IVoiceChannel channel2))
+            {
+                _logger.LogError("Konnte nicht den 2. Kanal als IVoiceChannel casten!");
```
Comment (blaumeise20) [inline DiscordBot.Modules/Services/YoutubeService.cs:55]: ...

File: DiscordBot.Modules/Services/YoutubeService.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                return;
+            }
+            if (_channel == null)
+            {
+                return;
+            }
+
+            if (!(_guild.GetChannel(youtubeSettings.ChannelID3) is IVoiceChannel channel3))
+            {
+                _logger.LogError("Konnte nicht den 3. Kanal als IVoiceChannel casten!");
```
Comment (blaumeise20) [inline DiscordBot.Modules/Services/YoutubeService.cs:65]: ...

File: DiscordBot.Modules/Services/YoutubeService.cs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+                var data = JsonConvert.DeserializeObject<Welcome>(response);
+                if (data.Items != null && data.Items.Length > 0)
+                {
+                    var subs = data.Items[0].Statistics.SubscriberCount;
+                    await channel.ModifyAsync(property =>
+                    {
+                        property.Name = $"Abonnenten: {subs}";
+                    });
+                    _logger.LogInformation($"Abonnenten eingeholt: der User mit der ID {youtubeSettings.YTChannelID} hat {subs} Abonnenten");
```
Comment (blaumeise20) [inline DiscordBot.Modules/Services/YoutubeService.cs:85]: ...

File: DiscordBot.Modules/Services/YoutubeService.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+                    var views = data.Items[0].Statistics.ViewCount;
+                    await channel2.ModifyAsync(property =>
+                    {
+                        property.Name = $"Views: {views}";
+                    });
+                    _logger.LogInformation($"Views eingeholt: der User mit der ID {youtubeSettings.YTChannelID} hat {views} Views");
```
Comment (blaumeise20) [inline DiscordBot.Modules/Services/YoutubeService.cs:91]: ...

File: DiscordBot.Modules/Services/YoutubeService.cs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+                    var user = data.Items[0].Snippet.Title;
+                    await channel3.ModifyAsync(property =>
+                    {
+                        property.Name = $"YouTube: {user}";
+                    });
+                    _logger.LogInformation($"YouTube Name eingeholt: der User mit der ID {youtubeSettings.YTChannelID} heißt {user}");
```
Comment (blaumeise20) [inline DiscordBot.Modules/Services/YoutubeService.cs:97]: ...

File: DiscordBot.Modules/Services/YoutubeService.cs [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+using DiscordBot.Domain.YouTubeAPI;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace DiscordBot.Modules.Services
+{
+    public class YoutubeService
+    {
+        private readonly YoutubeSettings youtubeSettings;
```
Comment (JKamsker) [inline DiscordBot.Modules/Services/YoutubeService.cs:18]: Naming convention for private fields: _myPrivateField

File: DiscordBot.Modules/Services/YoutubeService.cs [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+            }
+
+            if (!(_guild.GetChannel(youtubeSettings.ChannelID3) is IVoiceChannel channel3))
+            {
+                _logger.LogError("Konnte nicht den 3. Kanal als IVoiceChannel casten!");
+                return;
+            }
+
+
+            string query = "https://www.googleapis.com/youtube/v3/channels?part=snippet%2Cstatistics&id=" + youtubeSettings.YTChannelID + "&key=" + youtubeSettings.APIKey;
```
Comment (JKamsker) [inline DiscordBot.Modules/Services/YoutubeService.cs:70]: please use $"http...{youtubeSettings.YTChannelID }...."

File: DiscordBot.Modules/Services/YoutubeService.cs [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+            var _channel3 = _guild.GetChannel(youtubeSettings.ChannelID3) as IVoiceChannel;
+
+            if (_channel == null)
+            {
+                return;
+            }
+
+            if (!(_guild.GetChannel(youtubeSettings.ChannelID) is IVoiceChannel channel))
+            {
+                _logger.LogError("Konnte nicht den Kanal als IVoiceChannel casten!");
```
[the remark this answers is not in the document]
Comment (SIMULATAN) [reply DiscordBot.Modules/Services/YoutubeService.cs:45]: Me and @JKamsker said before your time, that we'll write all the Logs in German, also the Replys will be in German, only the GitHub Stuff is English

**Pull request closed without merging**